tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1373,6 +1373,45 @@ Status Converter::RenameAndMarkOutputTensors(
   return Status::OK();
 }
 
+#if IS_TRT_VERSION_GE(7, 1, 3, 0)
+// Force TensorRT deterministic behavior by enforcing
+// TRT tactic selection to be a specific one controlled by the env var:
+// TF_DEBUG_TRT_DETERMINISTIC_TACTIC_ID.
+class DeterministicAlgorithmSelector : public nvinfer1::IAlgorithmSelector
+{
+public:
+   //  Return value in [0, nbChoices] for a valid algorithm.
+   int32_t selectAlgorithms(const nvinfer1::IAlgorithmContext& algoContext,
+       const nvinfer1::IAlgorithm* const* algoChoices, int32_t nbChoices, int32_t* selection) override
+   {
+       // TensorRT always provides more than zero number of algorithms
+       // in selectAlgorithms.
+       assert(nbChoices > 0);
+
+       // retrieving the requested tactic ID
+       static int64 trt_tactic_id = [] {
+         int64 trt_tactic_id = -1;
+         TF_CHECK_OK(tensorflow::ReadInt64FromEnvVar(
+           "TF_DEBUG_TRT_DETERMINISTIC_TACTIC_ID",
+           /*default_val=*/0, &trt_tactic_id));
+         return trt_tactic_id;
+       }();
+
+       // making sure that the requested TRT Tactic ID doesn't go above the
+       // max value accepted.
+       selection[0] = std::min(static_cast<int32_t>(trt_tactic_id), nbChoices);
+       return 1;
```

Comment:
This block can be rewritten to:

selection[0] = std::min(algorithm_id_, nbChoices);
return 1;

Reply:
Done